### PR TITLE
Sgratzl/stratification

### PR DESCRIPTION
**Pull request by sgratzl**
see #199 

gif

### CODE REVIEW

File: src/SupportView.ts [hunk 1]
```diff
@@ -119,6 +119,9 @@ export default class SupportView extends EventHandler {
     this.datasets = convertTableToVectors(await listData())
       .filter((d) => d.idtypes.indexOf(this.idType) >= 0 && isPossibleDataset(d));
 
+    const matrixColumns = await Promise.all(this.datasets.filter((d) => d.desc.type === 'matrix').map(splitMatrixInVectors));
+    this.datasets.push(...[].concat(...matrixColumns));
```
Comment (sgratzl) [inline src/SupportView.ts]: `matrixColumns` has the type `IDataType[][]` using the first concat I create a flat array out of it `IDataType[]` which is later on pushed to `this.datasets`

File: src/SupportView.ts [hunk 2]
```diff
@@ -393,3 +395,99 @@ export function transposeMatrixIfNeeded(rowtype: IDType, d: IDataType) {
...
+    anyDesc.origin = matrix;
+    anyDesc.fqname = matrix.desc.fqname + '/' + colNames[i];
+    r.push(v);
+  }
+  return r;
+}
+
+function convertToTree(datasets: IDataType[]) {
+  // has a origin field, i.e. is derived from a matrix
+  const isDerivedDataset = (dataset: IDataType) => !!(<any>dataset.desc).origin;
```
Comment (sgratzl) [inline src/SupportView.ts]: ! ... check for not existence
!! ... check for existence and ensure it is a boolean

Comment (thinkh) [inline src/SupportView.ts]: Nice detail. Thanks for the explanation.

**Comment by sgratzl** [comment]
> I get thousands of requests to the same URL when adding a derived vector column of a matrix.

this specific bug was fixed by integrating the `dataset_fixes` branch. However, still plenty of requests are performed, since only the portion of the matrix is loaded that is needed at a time. Reason is probably how the sorting is implemented. 

p.s. don't forget to update phovea_server for a bugfix in this cornercase

**Comment by thinkh** [comment]
If a string column is added as first column then every row results in one request. Hence, I still see a lot of AJAX requests in the developer console. 

For later: Sorting might also be buggy and needs further testing. Please file a new issue.

**Pull request merged**